app/views/webshop/show.html.erb

```diff
@@ -1,4 +1,12 @@
 <h1>Welcome to Webshop</h1>
+
 <% @products.each do |product| %>
-  <%= "#{product.title} - #{product.description} - #{product.price}" %>
+<div id="product-<%= product.id %>">
+  <%= "#{product.title} - #{product.description} - #{product.price}" %><br/>
+    <% if @cart.nil? %>
```

Comment:
Where do you set the variable `@cart`? Wouldn't you be better off with checking the session for `[:cart_id]`?

If you actually have a `@cart` variable (that is actually holding an object of the `Cart` class) or if you check your session (as I mention above), you could do something like this (using a ternary operator) :
```
<%= link_to 'Add to cart', shopping_carts_path(product_id: product), method: @cart ? :PUT : :POST %>
```

And please skip the `<br/>`. It not necessary and shouldn't be used.

Reply:
I'll remove the <br/> and implement the ternary solution